### ISSUE TITLE
Native iOS Widget

### ISSUE THREAD

**Issue by jbobrow** [jbobrow/MTA-Subway-Now-Departing#2]
This would be quite nice for the following reasons:
1. [Dynamic Dates](https://developer.apple.com/documentation/widgetkit/displaying-dynamic-dates) allows for the widget to refresh without technically refreshing (i.e. updates in realtime)
2. Install can be directly from the AppStore (much less friction for casual users)
3. Setup for selecting train/station can be a much smoother and prettier flow (something like the following 3 steps)   

![Setup Flow](https://user-images.githubusercontent.com/772799/209875804-59ea2dec-09b1-4b1d-97f0-7e85f698366d.png)
4. Definitely not for the 💰, best as a free app
5. Enables WatchOS, could be pretty sweet on a watch face🤩 (credit to [Ace](https://www.linkedin.com/in/durjoy/))

**Comment by ignaspa** [comment]
Thanks for sharing this widget on scriptable! Was wondering if this native widget goal includes making the widget small enough for display on iOS lockscreen? The current scriptable widget understandably gets wrecked by the size & colour constraints.

**Comment by jbobrow** [comment]
@ignaspa fine suggestion. I hadn't considered it on the lock screen, but that sounds pretty handy. Have you tried it with scriptable?

**Comment by ignaspa** [comment]
Yeah, works normally on my homescreen but broken in both possible formats on the lockscreen.

**Comment by jbobrow** [comment]
Cool. I'm curious to know what you mean by broken. Can include a screenshot? I imagine for the lockscreen, the layout would need to be simplified much more.

**Comment by ignaspa** [comment]
![IMG_E75B8B9F1FC0-1](https://user-images.githubusercontent.com/29822176/212583852-88ddf456-e315-4e36-ab85-49e0f728929e.jpeg)
As you can see, both possible sizes of the homescreen widget type do not fully allow the scriptable UI to be visible. The line is visible but cut off, and the coloring looks less than clear.

**Comment by jbobrow** [comment]
Thanks for the visual. I see, it is a design/layout issue, not a technical one. Here's a branch that uses the relative time feature and then formats the widget for your homescreen. I titled this script "MTA-Homescreen" in my Scriptable apps and added this one to my homescreen. It looks like the following.
![IMG_86B8E5EE18B5-1](https://user-images.githubusercontent.com/772799/212723075-741c6857-5e31-45c5-bc1d-35364d940c05.jpeg)
Let me know how that works for you.

**Comment by ignaspa** [comment]
Looks decent in the bigger widget format, thanks!
